clients/drcachesim/tests/decode_cache_test.cpp

```diff
@@ -262,29 +298,47 @@ int
 test_main(int argc, const char *argv[])
 {
     void *drcontext = dr_standalone_init();
-    std::string err = check_decode_caching(drcontext, /*persist_instrs=*/false,
-                                           /*use_module_mapper=*/false);
+    std::string err = check_decode_caching(drcontext, /*use_module_mapper=*/false,
+                                           /*include_decoded_instr=*/false,
+                                           /*persist_decoded_instr=*/false);
     if (err != "") {
         std::cerr << err << "\n";
         exit(1);
     }
-    err = check_decode_caching(drcontext, /*persist_instrs=*/true,
-                               /*use_module_mapper=*/false);
+    err = check_decode_caching(drcontext, /*use_module_mapper=*/false,
+                               /*include_decoded_instr=*/true,
+                               /*persist_decoded_instr=*/false);
+    if (err != "") {
+        std::cerr << err << "\n";
+        exit(1);
+    }
+    err = check_decode_caching(drcontext, /*use_module_mapper=*/false,
+                               /*include_decoded_instr=*/true,
+                               /*persist_decoded_instr=*/true);
     if (err != "") {
         std::cerr << err << "\n";
         exit(1);
     }
 #ifndef WINDOWS
     // TODO i#5960: Enable these tests after the test-only Windows issue is
     // fixed.
-    err = check_decode_caching(drcontext, /*persist_instrs=*/false,
-                               /*use_module_mapper=*/true);
+    err = check_decode_caching(drcontext, /*use_module_mapper=*/true,
+                               /*include_decoded_instr=*/false,
+                               /*persist_decoded_instr=*/false);
+    if (err != "") {
+        std::cerr << err << "\n";
+        exit(1);
+    }
+    err = check_decode_caching(drcontext, /*use_module_mapper=*/true,
+                               /*include_decoded_instr=*/true,
+                               /*persist_decoded_instr=*/false);
     if (err != "") {
         std::cerr << err << "\n";
         exit(1);
     }
-    err = check_decode_caching(drcontext, /*persist_instrs=*/true,
-                               /*use_module_mapper=*/true);
+    err = check_decode_caching(drcontext, /*use_module_mapper=*/true,
+                               /*include_decoded_instr=*/true,
```

Comment:
Is it illegal to have include=false persist=true?

Reply:
Yes it is illegal. It's included in the doc, and there's also an assert.